client/client-multi/client-c/src/kaa.c

```diff
@@ -80,19 +85,25 @@ static kaa_error_t kaa_context_create(kaa_context_t **context_p, kaa_logger_t *l
 
     (*context_p)->logger = logger;
 
-    kaa_error_t error = kaa_status_create(&((*context_p)->status));
+    kaa_error_t error = KAA_ERR_NONE;
+    (*context_p)->status = (kaa_status_holder_t *) KAA_MALLOC(sizeof(kaa_status_holder_t));
+    if (!(*context_p)->status)
+        error = KAA_ERR_NOMEM;
+
+    if (!error)
```

Comment:
This code is used to free resources in case of initialization errors. You can find it at the end of kaa_context_create function

    if (error) {
        kaa_context_destroy(*context_p);
        *context_p = NULL;
    }

Hence, according to current implementation I couldn't use KAA_RETURN_IF_ERR here.